File: NERP.Common/Models/Geometry.cs
using System.Collections.Generic;

namespace NERP.Common.Models
{
    public class Point
    {
        public string type { get; set; } = "Point";

        public double[] coordinates { get; set; }

    }

    public class MultiPoint
    {
        public string type { get; set; } = "MultiPoint";

        public IEnumerable<double[]> coordinates { get; set; }

    }

    public class LineString
    {
        public string type { get; set; } = "LineString";

        public IEnumerable<double[]> coordinates { get; set; }

    }

    public class MultiLineString
    {
        public string type { get; set; } = "MultiLineString";

        public IEnumerable<IEnumerable<double[]>> coordinates { get; set; }

    }

    public class Polygon
    {
        public string type { get; set; } = "Polygon";

        public IEnumerable<IEnumerable<double[]>> coordinates { get; set; }
        //public GeoJsonPolygon<GeoJson2DGeographicCoordinates> Coordinates { get; set; }

    }

    public class MultiPolygon
    {
        public string type { get; set; } = "MultiPolygon";

        public IEnumerable<IEnumerable<IEnumerable<double[]>>> coordinates { get; set; }

    }

}

File: NERP.Admin.Web1/ClientApp/src/app/common/geo/geometry.ts
export type Position = number[];

export class Point {
    type: string = "Point";
    coordinates: Position

    constructor(coordinates?: Position) {
        this.coordinates = coordinates;
    }
}

export class MultiPoint {
    type: string = "MultiPoint";
    coordinates: Array<Position>;

    constructor(coordinates?: Array<Position>) {
        this.coordinates = coordinates;
    }
}

export class LineString {
    type:string = "LineString";
    coordinates: Array<Position>;

    constructor(coordinates?: Array<Position>) {
        this.coordinates = coordinates;
    }
}

export class MultiLineString {
    type: string = "MultiLineString";
    coordinates: Array<Array<Position>>;
}

export class Polygon {
    type: string = "Polygon";
    coordinates: Array<Array<Position>>;

    constructor(coordinates?: Array<Array<Position>>) {
        this.coordinates = coordinates;
    }
}

export class MultiPolygon {
    type: string = "MultiPolygon";
    coordinates: Array<Array<Array<Position>>>;

    constructor(coordinates?: Array<Array<Array<Position>>>) {
        this.coordinates = coordinates;
    }
}

//export enum GeoJsonObjectType {
//    Feature = 0,
//    FeatureCollection = 1,
//    GeometryCollection = 2,
//    LineString = 3,
//    MultiLineString = 4,
//    MultiPoint = 5,
//    MultiPolygon = 6,
//    Point = 7,
//    Polygon = 8
//}
